site/gatsby-site/server/emails/index.ts

```diff
@@ -0,0 +1,95 @@
+import { EmailParams, MailerSend, Recipient } from "mailersend";
+import config from "../config";
+import templates from "./templates";
+import * as reporter from '../reporter';
+import assert from "assert";
+
+interface SendEmailParams {
+    recipients: {
+        email: string;
+        userId: string;
+    }[];
+    subject: string;
+    dynamicData: {
+        incidentId?: string;
+        incidentTitle?: string;
+        incidentUrl?: string;
+        incidentDescription?: string;
+        incidentDate?: string;
+        developers?: string;
+        deployers?: string;
+        entitiesHarmed?: string;
+        reportUrl?: string;
+        reportTitle?: string;
+        reportAuthor?: string;
+        entityName?: string;
+        entityUrl?: string;
+    };
+    templateId: string;
+}
+
+export const replacePlaceholdersWithAllowedKeys = (template: string, data: { [key: string]: string }, allowedKeys: string[]): string => {
+    return template.replace(/\{\{\s*(\w+)\s*\}\}/g, (match, key) => {
+        return allowedKeys.includes(key) && key in data ? data[key] : match;
+    });
+}
+
+export const mailersendBulkSend = async (emails: EmailParams[]) => {
+
+    const mailersend = new MailerSend({
+        apiKey: config.MAILERSEND_API_KEY,
+    });
+
+    assert(emails.every(email => email.to.length == 1), 'Emails must have exactly one recipient');
```

Comment:
This is I would extend this to handle both `to` + `cc` <= 1. (or == 1).

thoughts?